modules/programs/firefox.nix

```diff
@@ -304,16 +370,54 @@ in
           };
 
         "${profilesPath}/${profile.path}/user.js" =
-          mkIf (profile.settings != {} || profile.extraConfig != "") {
-            text = mkUserJs profile.settings profile.extraConfig;
+          mkIf (profile.settings != {} || profile.extraConfig != "" || profile.extensions != [] || profile.extraPolicies != {}) {
+            text = let
+              settings = if (profile.extensions != [] || profile.extraPolicies != {})
+              then profile.settings // (if (profile.individualPolicies == true) then { "toolkit.policies.loadFrom" = 2; } else { "toolkit.policies.perUserDir" = true; } )
+              else profile.settings;
+            in mkUserJs settings profile.extraConfig;
           };
 
-        "${profilesPath}/${profile.path}/extensions" = mkIf (cfg.extensions != []) {
-          source = "${extensionsEnvPkg}/share/mozilla/${extensionPath}";
-          recursive = true;
-          force = true;
+        "${profilesPath}/${profile.path}/policies.json" = mkIf (profile.individualPolicies == true && (profile.extraPolicies != {} || profile.extensions != [])) {
+          text = (mkPolicies profile);
         };
       })
     );
+
+    systemd.user = {
+      services = {
+        firefox-policies-writer = {
+          Unit = { Description = "Firefox Policies Writer"; };
+
+          Install = { WantedBy = [ "default.target" ]; };
+
+          Service = {
+            Type = "oneshot";
+            RemainAfterExit = true;
+
+            ExecStart = let
+              policiesFile = pkgs.writeText "policies.json" (mkPolicies cfg);
+            in (toString (pkgs.writeShellScript "write-firefox-policies-file" ''
+              set -xe
+
+              # set our runtime dir + policies location here 
+              XDG_RUNTIME_DIR="/run/user/$($DRY_RUN_CMD  ${pkgs.coreutils}/bin/id -u)"
```

Comment:
`$DRY_RUN_CMD` won't be defined inside this shell script, it's only for activation scripts. So you can remove it.

Reply:
Fixed in newest `update` commit. Will squash them once everything is good-to-go :)